uuid.go

```diff
@@ -344,6 +345,29 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	if bytes.Equal(b, []byte("null")) {
+		u.UUID, u.Valid = Nil, false
+		return nil
+	}
+
+	u.Valid = true
```

Comment:
Wouldn't it make more sense to do `u.Valid = true` after `if err := json.Unmarshal(b, &u.UUID); err != nil { ... }`? What's the reason you decided to put it on top, instead of below?